test/component/custodian/state-shift.test.js

```diff
@@ -37,99 +32,43 @@ test('custodian state shall shifts', t => {
-  t.true(custodian.isPrepared(), 'custodian shall be logged in')
-  t.true(custodian.isLoggedIn(), 'custodian shall be logged in')
-  t.false(custodian.isNetworkUnavailable())
+  /* Replace flora with NetworkMock */
+  custodian.networkAgent._flora = new NetworkMock()
 
-  custodian.onNetworkDisconnect()
-  t.false(custodian.isRegistering())
-  t.false(custodian.isPrepared())
-  t.true(custodian.isLoggedIn(), 'custodian shall be still logged in')
-  t.true(custodian.isNetworkUnavailable(), 'custodian shall be network unavailable on network disconnected')
-})
+  /* Mock bluetooth */
+  custodian.initBluetooth()
+  custodian.bluetoothStream.disconnect()
+  custodian.openBluetooth = () => {}
+  custodian.closeBluetooth = () => {}
 
-test('custodian shall start network app on network disconnect if not logged in', t => {
-  t.plan(4)
-  mock.mockReturns(wifi, 'getNumOfHistory', 0)
+  /* test start */
+  t.true(custodian.isUnconfigured())
+  custodian.configureNetwork()
+  t.true(custodian.isConfiguringNetwork())
 
-  var runtime = {
-    resetNetwork: function () {
-      t.pass('onNetworkDisconnect shall trigger runtime#resetNetwork')
-    },
-    component: {
-      appScheduler: {
-        appMap: {}
-      },
-      lifetime: {
-        getCurrentAppId: () => undefined
-      },
-      light: {
-        stop: function () {}
-      },
-      wormhole: {
-        setOffline: function () {
-          t.fail('onNetworkDisconnect shall not trigger runtime.wormhole#setOffline')
-        }
-      }
+  custodian.bluetoothStream.emit('data', {
+    topic: 'bind',
+    data: {
+      U: 'test_ssid',
+      P: 'passwd'
     }
-  }
-  var custodian = new Custodian(runtime)
-  custodian.onNetworkDisconnect()
-  t.false(custodian.isRegistering())
-  t.false(custodian.isPrepared())
-  t.true(custodian.isNetworkUnavailable())
-})
+  })
+  t.true(custodian.isConnecting())
+  custodian.networkAgent._flora.subscribe(
+    'network.status',
+    custodian.networkAgent._handleNetworkStatus.bind(custodian.networkAgent))
 
-test('custodian shall reset network', t => {
-  t.plan(10)
-  var runtime = {
-    reconnect: function () {
-      t.fail('onNetworkConnect shall not trigger runtime#reconnect')
-    },
-    openUrl: function () {
-      t.pass('resetNetwork shall trigger runtime#startApp')
-    },
-    component: {
-      appScheduler: {
-        appMap: {}
-      },
-      lifetime: {
-        getCurrentAppId: () => undefined
-      },
-      light: {
-        stop: function () {}
-      },
-      wormhole: {
-        setOffline: function () {
-          t.pass('resetNetwork shall trigger runtime.wormhole#setOffline')
-        }
-      }
-    }
-  }
-  var custodian = new Custodian(runtime)
-  custodian.onNetworkConnect()
-  custodian.onLoggedIn()
-  t.false(custodian.isRegistering())
-  t.true(custodian.isPrepared(), 'custodian shall be prepared')
-  t.false(custodian.isNetworkUnavailable())
+  setTimeout(() => {
+    t.true(custodian.isLogging())
```

Comment:
Naming issue: `Logging` might not be proper to describe login/authorization process.

Reply:
How about isLoggingIn?

Reply:
Though the word `log in` may be exactly what it's doing now, I'd recommend change the name to `authorization` for convenient customization.